include/circt/Dialect/Handshake/HandshakeOps.td

```diff
@@ -137,7 +137,6 @@ def InstanceOp : Handshake_Op<"instance", [CallOpInterface]> {
       $_state.addOperands(operands);
       $_state.addAttribute("module", SymbolRefAttr::get(module));
       $_state.addTypes(module.getResultTypes());
-      $_state.addTypes({$_builder.getType<::mlir::NoneType>()});
```

Comment:
@Dinistro `ArrayRef<Value>` -> `ValueRange` seems like a great change, but why is this line being removed? This seems unrelated and potentially load bearing... I'm surprised no tests failed.

Reply:
This change should have been a separate commit, but I seem to have accidentally added that here. 
The builder is in fact untested. Note that the provided `FuncOp` should already have the correct result types, including the `NoneType` for the control signal. Therefore, adding another `NoneType` at the end makes this invalid. 

In my out-of-tree work, that change fixed the usage of the builder.

Should I maybe add additional tests for that change?

Reply:
I see, well first off thank you for fixing this. In the future, please try to keep patches like this separate. And yes, please add a test to cover the issue (both for this fix here, and any patches moving forward, even if they are similarly small).

Reply:
I'll do so in the future. Sorry for missing it here. 

How does one write tests for operation builders, with unit-tests?

Reply:
I found the bug while creating instances in my out-of-tree work. 

The `StandardToHandshake` pass only creates `InstanceOp`s when it transforms `func::CallOp`s. At this point, the builder with the `SymbolRefAttr` is used. This builder is not used anywhere in CIRCT it seems.

Reply:
> This builder is not used anywhere in CIRCT it seems.

I see, well in this case I don't think we need to write a test pass just to exercise it. Thanks for being diligent.